qabelbox/src/main/java/de/qabel/qabelbox/communication/BaseServer.java

```diff
@@ -2,32 +2,75 @@
 
 import android.util.Log;
 
-import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.util.LinkedList;
+import java.util.List;
 import java.util.Locale;
 import java.util.concurrent.TimeUnit;
 
+import de.qabel.qabelbox.QabelBoxApplication;
+import de.qabel.qabelbox.communication.callbacks.RequestCallback;
+import de.qabel.qabelbox.communication.connection.ConnectivityManager;
+import okhttp3.Call;
 import okhttp3.MediaType;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 
-/**
- * Created by danny on 11.02.16.
- */
 public class BaseServer {
 
     protected final OkHttpClient client;
     protected final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
     private static final String TAG = "BaseServer";
     URLs urls;
 
+    private ConnectivityManager connectivityManager;
+    private List<RequestAction> requestActionQueue = new LinkedList<>();
+
     /**
      * create new instance of http client and set timeouts
      */
     public BaseServer() {
 
+        connectivityManager = new ConnectivityManager(QabelBoxApplication.getInstance().getApplicationContext());
```

Comment:
If you need a context here, inject it please. No need to use the static method in QabelBoxApplication. That method always provokes NullpointerExceptions.